kubernetes/data_source_kubernetes_endpoints.go

```diff
@@ -0,0 +1,35 @@
+package kubernetes
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesEndpoint() *schema.Resource {
```

Comment:
Please add a `V1` suffix to the function name and make it plural to align with `kind` and `apiVersion` -- `dataSourceKubernetesEndpointsV1`.